app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt

```diff
@@ -18,23 +19,52 @@ import org.json.JSONObject
 
 class MosaicRepository(application: Application) {
 
-    val TAG = MosaicRepository::class.simpleName
+    private val TAG = MosaicRepository::class.simpleName
 
-    private val drawables = mutableListOf<Drawable>()
+    private val drawables = mutableListOf<InstagramImage>()
     private val mosaicsList = MutableLiveData<List<Drawable>>()
     fun getMosaicsList(): LiveData<List<Drawable>> {
         return mosaicsList
     }
 
+    private val instagramUserName = MutableLiveData<String>()
+    fun getUserName(): LiveData<String> {
+        return instagramUserName
+    }
+
     init {
         if (isNetworkConnected(application)) {
             Log.d(TAG, "Successful internet connection")
+            val userName = Settings.getUserName(application)
+            if (userName == null) {
+                loadInstagramUserName(application, userName)
+            } else {
+                instagramUserName.postValue(userName!!)
+                loadInstagramUserName(application, userName)
+            }
             loadImagesFromInstagram(application)
         } else {
             Log.d(TAG, "No internet connection or network problems")
         }
     }
 
+    private fun loadInstagramUserName(application: Application, userName: String?) {
+        val token = AccessToken.getCurrentAccessToken()
+        GraphRequest(
+            token,
+            "/$INSTAGRAM_ID?fields=username",
+            null,
+            HttpMethod.GET
+        ) { findUserNameResponse ->
+            val userNameResponse: JSONObject = findUserNameResponse.jsonObject
+            if (userName != userNameResponse.getString("username")) {
+                instagramUserName.postValue(userNameResponse.getString("username"))
```

Comment:
Please move serNameResponse.getString("username") to variable

Reply:
done